modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/XmlProcessor.java

```diff
@@ -0,0 +1,889 @@
+        }
+
+        private String getAttributeName(String ignoredUri, String localName, String qName) {
+            String attrName;
+            if (removeNamespaces) {
+                attrName = localName != null && localName.isEmpty() == false ? localName : qName;
+            } else {
+                attrName = qName;
+            }
+
+            // Apply toLower if enabled
+            if (toLower) {
+                attrName = attrName.toLowerCase(Locale.ROOT);
+            }
+
+            return attrName;
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured SAX parser factory for XML parsing using XmlUtils.
+     */
+    private static SAXParserFactory createSecureSaxParserFactory() {
+        try {
+            SAXParserFactory factory = XmlUtils.getHardenedSaxParserFactory();
+            factory.setValidating(false);
+            factory.setNamespaceAware(false);
+            return factory;
+        } catch (Exception e) {
+            logger.warn("Cannot configure secure XML parsing features - XML processor may not work correctly", e);
+            return null;
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured namespace-aware SAX parser factory for XML parsing using XmlUtils.
+     */
+    private static SAXParserFactory createSecureSaxParserFactoryNamespaceAware() {
+        try {
+            SAXParserFactory factory = XmlUtils.getHardenedSaxParserFactory();
+            factory.setValidating(false);
+            factory.setNamespaceAware(true);
+            return factory;
+        } catch (Exception e) {
+            logger.warn("Cannot configure secure namespace-aware XML parsing features - XML processor may not work correctly", e);
+            return null;
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured DocumentBuilderFactory for DOM creation using XmlUtils.
+     * Since we only use this factory to create empty DOM documents programmatically
+     * (not to parse XML), we use the hardened builder factory.
+     * The SAX parser handles all XML parsing with appropriate security measures.
+     */
+    private static DocumentBuilderFactory createSecureDocumentBuilderFactory() {
+        try {
+            DocumentBuilderFactory factory = XmlUtils.getHardenedBuilderFactory();
+            factory.setValidating(false); // Override validation for DOM creation
+            return factory;
+        } catch (Exception e) {
+            logger.warn("Cannot configure secure DOM builder factory - XML processor may not work correctly", e);
+            return null;
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured XPath object expression evaluation using XmlUtils.
+     */
+    private static XPath createSecureXPath() {
+        try {
+            return XmlUtils.getHardenedXPath();
+        } catch (Exception e) {
+            logger.warn("Cannot configure secure XPath object - XML processor may not work correctly", e);
+            return null;
```

Comment:
I followed the existing pattern of returning null here, but honestly I think we should blow up and die much more aggressively if this happens. The processor is just going to throw NPEs at every step of the way anyway, and that seems... unproductive.